src/malli/transform.cljc

```diff
@@ -158,4 +163,8 @@
 (defn json-transformer [schema]
   (get +json-decoders+ (m/name schema)))
 
+(defn strip-extra-keys-transformer [schema]
+  (when-some [keys (-> schema m/properties ::m/map-keys)]
+    (partial strip-extra-keys keys)))
```

Comment:
instead of partial function, we could (re-)parse the schema and use it to create a 1-arity function which sees the keys via a closure. It more efficient than partial function (which creates an extra wrapper fn).